onnxruntime/core/session/abi_session_options.cc

```diff
@@ -20,6 +20,40 @@ OrtSessionOptions::OrtSessionOptions(const OrtSessionOptions& other)
     : value(other.value), provider_factories(other.provider_factories) {
 }
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
+onnxruntime::Status OrtSessionOptions::RegisterCustomOpsLibrary(std::string library_name) {
+  const auto& platform_env = onnxruntime::Env::Default();
+  void* library_handle = nullptr;
+
+  ORT_RETURN_IF_ERROR(platform_env.LoadDynamicLibrary(library_name, false, &library_handle));
+  if (!library_handle) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to load dynamic library ", library_name);
+  }
+
+  OrtStatus*(ORT_API_CALL * RegisterCustomOps)(OrtSessionOptions * options, const OrtApiBase* api) = nullptr;
+  ORT_RETURN_IF_ERROR(platform_env.GetSymbolFromLibrary(library_handle, "RegisterCustomOps",
+                                                        (void**)&RegisterCustomOps));
+
+  // Call the exported RegisterCustomOps function and store the return value in a unique_ptr.
+  const std::unique_ptr<OrtStatus, decltype(&OrtApis::ReleaseStatus)> status(RegisterCustomOps(this, OrtGetApiBase()),
```

Comment:
TODO: Add a test to `test_inference.cc` that exercises a failed called to a custom op library's `RegisterCustomOps` function.

Reply:
Done